packages/dom-adapters/src/FormattingAdapter/index.ts

```diff
@@ -68,17 +72,62 @@ export class FormattingAdapter {
       if (selection) {
         const range = selection.getRangeAt(0);
 
-        const inlineElement = tool.createWrapper();
+        const inlineElement = tool.createWrapper(event.detail.data.data);
+
+        const extracted = range.extractContents();
 
         /**
          * Insert contents from range to new inline element and put created element in range
          */
-        inlineElement.appendChild(range.extractContents());
+        inlineElement.appendChild(extracted);
+
         range.insertNode(inlineElement);
       }
     }
   }
 
+  /**
+   * Allows to render formatting inside a passed input
+   *
+   * @param input - input element to apply format to
+   * @param inlineFragment - instance that contains index, toolName and toolData
+   * @param inlineFragment.index - text range inside of the input element
+   * @param inlineFragment.toolName - name of the tool, which format to apply
+   * @param inlineFragment.toolData - additional data for the tool
+   */
+  public formatElementContent(input: HTMLElement, inlineFragment: InlineFragment): void {
+    const toolName = inlineFragment.tool;
+    const toolData = inlineFragment.data;
+    const index = inlineFragment.range;
+
+    const tool = this.#tools.get(toolName);
+
+    if (tool === undefined) {
+      throw new Error(`FormattingAdapter: tool ${toolName} is not attached`);
+    }
+
+    const [start, end] = index;
+
+    /**
+     * Create range with positions specified in index
+     */
+    const range = document.createRange();
+
+    range.setStart(input, start);
+    range.setEnd(input, end);
+
+    const inlineElement = tool.createWrapper(toolData);
+
+    const extracted = range.extractContents();
+
+    /**
+     * Insert contents from range to new inline element and put created element in range
+     */
+    inlineElement.appendChild(extracted);
+
+    range.insertNode(inlineElement);
```

Comment:
This could be extracted to a separate method to re-use in #handleModelUpdate method as well